src/main/java/core/basesyntax/model/OperationType.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.model;
+
+public enum OperationType {
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+    private String code;
+
+    OperationType(String code) {
+        this.code = code;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public static OperationType fromString(String text) {
+        for (OperationType type : OperationType.values()) {
+            if (type.code.equalsIgnoreCase(text)) {
+                return type;
+            }
+        }
+        return null;
```

Comment:
it's generally preferred to throw an exception (e.g. IllegalArgumentException) than return a null silently if that's now an expected behavior